### ISSUE TITLE
pack from .csproj and .nuspec content regression from 3.3

### ISSUE THREAD

**Issue by maxgax** [NuGet/Home#2658]
I generate NuGet packages as part of my build process, using the NuGet pack command. After upgrading from 3.3.0 to 3.4.3, I noticed that my packages no longer included content files which were previously being added to my packages.

I am currently using "dummy projects" which in turn have references to multiple projects. I am only using the nuspec file for metadata and depend on references to pull in all of the necessary files. Previously I was able to pull in all of the binaries, content files, and package.config dependencies of all referenced projects just by running

> NuGet.exe pack PathToDummyProject.csproj -IncludeReferencedProjects -Version 1.0.0.0 -Properties Configuration=Release

Upon upgrading to 3.4.3, the pack command still succeeds but no longer adds any content files to the package.

The content files are listed in the .csproj files like so. . .

```
<ItemGroup>
    <Content Include="Install\Descriptors\Configuration.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
    </Content>
  </ItemGroup>

```

I haven't been able to find any documentation that lends to the cause of this issue. Any insight would be appreciated.

**Comment by toddm** [comment]
Sorry you're having trouble.  I'm trying to repro the problem but so far I don't see a difference between 3.3.0 and 3.4.3. I might not have my test project set up the same way as you, though. Can you give us a project to use to see the issue? (please email projects to support@nuget.org)

**Comment by maxgax** [comment]
Here is a small solution with two projects that reproduces the issue. The "DummyProject" is a Console application project which references the Class Library project "ReferencedProject". I've specified a nuspec file for DummyProject.csproj that only contains metadata. I've shown below the results of creating the package using both NuGet version 3.3.0 and 3.4.3. My expectation is that version 3.4.3 would behave the same as 3.3.0 by including the content in addition to the assemblies of the referenced project in the created package.

[NuGetBugRepro.zip](https://github.com/NuGet/Home/files/237919/NuGetBugRepro.zip)

**Using NuGet version 3.3.0**

> D:\NuGetBugRepro.nuget>NuGet.exe pack ..\DummyProject\DummyProject.csproj -IncludeReferencedProjects -Properties Configuration=Debug -Verbosity detailed

**The Result (Version 3.3.0)**

> MSBuild auto-detection: using msbuild version '14.0' from 'C:\Program Files (x86)\MSBuild\14.0\bin'. Use option -MSBuildVersion to force nuget to use a specific version of MSBuild.
> Attempting to build package from 'DummyProject.csproj'.
> Packing files from 'D:\NuGetBugRepro\DummyProject\bin\Debug'.
> Using 'DummyProject.nuspec' for metadata.
> Add file 'D:\NuGetBugRepro\DummyProject\bin\Debug\DummyProject.exe' to package as 'lib\net452\DummyProject.exe'
> Add file 'D:\NuGetBugRepro\ReferencedProject\bin\Debug\ReferencedProject.dll' to package as 'lib\net452\ReferencedProject.dll'
> Add file 'D:\NuGetBugRepro\ReferencedProject\Content\contentFile.json' to package as 'content\Content\contentFile.json'
> Id: DummyProject
> Version: 1.0.0
> Authors: Team
> Description: Reproduction of NuGet bug
> Tags: Repro
> Dependencies: None
> Added file 'content\Content\contentFile.json'.
> Added file 'lib\net452\DummyProject.exe'.
> Added file 'lib\net452\ReferencedProject.dll'.
> Successfully created package 'D:\NuGetBugRepro.nuget\DummyProject.1.0.0.nupkg'.

**Contents of package created (Version 3.3.0)**
[DummyProject.1.0.0.zip](https://github.com/NuGet/Home/files/237931/DummyProject.1.0.0.zip)

**Update NuGet**

> D:\NuGetBugRepro.nuget>NuGet.exe update -self
> Checking for updates from https://www.nuget.org/api/v2/.
> Currently running NuGet.exe 3.3.0.
> Updating NuGet.exe to 3.4.3.
> Update successful.

**Using NuGet version 3.4.3**

> D:\NuGetBugRepro.nuget>NuGet.exe pack ..\DummyProject\DummyProject.csproj -IncludeReferencedProjects -Properties Configuration=Debug -Verbosity detailed

**The Result (Version 3.4.3)**

> Attempting to build package from 'DummyProject.csproj'.
> MSBuild auto-detection: using msbuild version '14.0' from 'C:\Program Files (x86)\MSBuild\14.0\bin'. Use option -MSBuildVersion to force nuget to use a specific version of MSBuild.
> Packing files from 'D:\NuGetBugRepro\DummyProject\bin\Debug'.
> Using 'DummyProject.nuspec' for metadata.
> Add file 'D:\NuGetBugRepro\DummyProject\bin\Debug\DummyProject.exe' to package as 'lib\net452\DummyProject.exe'
> Add file 'D:\NuGetBugRepro\ReferencedProject\bin\Debug\ReferencedProject.dll' to package as 'lib\net452\ReferencedProject.dll'
> Id: DummyProject
> Version: 1.0.0
> Authors: Team
> Description: Reproduction of NuGet bug
> Tags: Repro
> Dependencies: None
> Added file 'lib\net452\DummyProject.exe'.
> Added file 'lib\net452\ReferencedProject.dll'.
> Successfully created package 'D:\NuGetBugRepro.nuget\DummyProject.1.0.0.nupkg'.

**Contents of package created (Version 3.4.3)**
[DummyProject.1.0.0.zip](https://github.com/NuGet/Home/files/237951/DummyProject.1.0.0.zip)

**Comment by toddm** [comment]
Thanks! I am able to repro the problem. I'll start debugging it.

**Comment by toddm** [comment]
The fix is in the Dev branch as well as release-3.4.4.

**Comment by rrelyea** [comment]
@maxgax - can you please verify the fix:
1. download not yet final version 3.4.4. of nuget.commandline.nupkg from https://www.myget.org/F/nugetbuild/api/v2/package/NuGet.CommandLine/3.4.4-rtm-870
2. rename the .nupkg to .zip
3. Grab nuget.exe from inside the zip file in the tools directory.
4. Try this version of nuget.exe to verify that it has fixed your issue.
5. Tell us how it went

**Comment by yishaigalatzer** [comment]
Reopening for validation - Please include the commit ids

**Comment by toddm** [comment]
https://github.com/NuGet/NuGet.Client/commit/c2443fd2aec84a1c9b1ab1b34a4abe79dd3eb1fa https://github.com/NuGet/NuGet.Client/commit/10a986f0959638cf975688619b720a1a0eec193a

**Comment by alpaix** [comment]
Was able to repro using provided project.
Validated the fix with 3.4.4.870 build.
Closing...
